File: Model/Player/PBuilder.kt
package Model.Player

import Model.Location
import Model.Move
import Model.Map
import Model.Neutral.Builder

class PBuilder(location: Location) : Builder() {
    override var loc = location
    override var use = true
    override var drive = true
    override var owner = 1

    override fun isValidMove(move: Move, map: Map) : Boolean {
        return true
    }
}

File: typescript/src/PlanetTS1/Model/Player/Units/PBuilder.ts
import { ALocation as ALocation } from "../../Utilities"

export class PBuilder extends Builder {
    loc : ALocation
    use : boolean
    drive : boolean
    owner : number
    icon : string

    constructor(location : ALocation) {
        super()
        this.loc = location
        this.use = true
        this.drive = true
        this.owner = 1
        this.icon = "pB"
    }
}